### PR TITLE
[#36][#38] [iOS] [UI] [Integrate] As a user, I can see the survey detail

### LINKED ISSUE #36: [iOS] [UI] As a user, I can see the survey detail
## Why

Before taking a survey, users should be able to see the full preview of the details to decide whether the survey matches their expectations.

## Acceptance Criteria

- Display the back button
- Display the Title label
   - No maximum number of lines
- Display the Description label
   - No maximum number of lines
- The screen is scrollable only when the content does not fit the screen

## Design

[Survey](https://www.figma.com/proto/GjRPOjDyZ6f4EDL3wKarRK/Challenge-Mobile-App?page-id=0%3A1&node-id=34%3A868&scaling=min-zoom)

### LINKED ISSUE #38: [iOS] [Integrate] As a user, I can see the survey detail
## Why

Before taking a survey, users should be able to see the full preview of the details so that they can decide whether or not the survey matching their expectation.

## Acceptance Criteria

- Upon tapping the Survey Disclosure button in Home, navigate the user to Survey Details with the selected survey.
    - Pass and use the selected survey data fetched in Home.
- Use the survey title as the Title label
- Use the survey description as the Description label
- Use the survey cover image URL as the background image
    - The image is zoomed to 120% anchoring to the top right of the screen

## Design

[Survey](https://www.figma.com/proto/GjRPOjDyZ6f4EDL3wKarRK/Challenge-Mobile-App?page-id=0%3A1&node-id=34%3A868&scaling=min-zoom)

### PR DESCRIPTION
**Pull request by suho**
- Resolves #36
- Resolves #38

## What happened 👀

Added survey detail screen for the iOS app.

## Insight 📝

I just created a new survey detail screen also with UI tests 

## Proof Of Work 📹

https://user-images.githubusercontent.com/19943832/209612303-0a0b221d-2ef2-463c-b124-83b10e2ce393.mp4

### CODE REVIEW

File: ios/Surveys/Sources/Presentation/Modules/SurveyDetail/SurveyDetailView.swift [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                    Asset.backIcon.image
+                }
+            }
+        }
+        .navigationBarBackButtonHidden(true)
+    }
+
+    init(viewModel: SurveyDetailViewModel) {
+        self.viewModel = viewModel
+    }
```
Comment (nkhanh44) [inline ios/Surveys/Sources/Presentation/Modules/SurveyDetail/SurveyDetailView.swift:83]: I think SurveyDetailView has its own initializer with a param `viewModel`, so we can safely remove this. Ignore this suggestion if you have future plan for this initializer 🙏 .

**Review by nkhanh44 — COMMENTED** [review]
Really nice work! Just a few concerns.
> The image is zoomed to 120% anchoring to the top right of the screen

Based on [AC #38](https://github.com/suho/kmm-ic/issues/38), maybe we forget the zooming 120% attribute?

File: ios/Surveys/Sources/Presentation/Modules/SurveyDetail/SurveyDetailView.swift (hunk 1 above)
Comment (suho) [reply ios/Surveys/Sources/Presentation/Modules/SurveyDetail/SurveyDetailView.swift]: I fixed both zooming 120% and remove the init here: https://github.com/suho/kmm-ic/pull/127/commits/df14905597ce00e93dbe116799f136dfb7c87533

Please note for custom transition of swiftui views, we might need to do it in another story

**Review by nkhanh44 — APPROVED** [review]
🚀

File: ios/Surveys/Sources/Presentation/Modules/Home/HomeViewModel.swift [hunk 2]
```diff
@@ -22,8 +22,22 @@ final class HomeViewModel: ObservableObject {
...
+    var surveyDetailViewModel: SurveyDetailViewModel {
+        let survey = surveys[currentPage]
+        let surveyArgument = SurveyArgument(
+            id: survey.id,
+            title: survey.title,
+            description: survey.description_,
+            imageURLString: survey.coverImageUrl
+        )
+        return SurveyDetailViewModel(survey: surveyArgument)
+    }
```
Comment (luongvo) [inline ios/Surveys/Sources/Presentation/Modules/Home/HomeViewModel.swift:39]: I might not be familiar with the ios project template. However, from the system design, does it make sense to build the new SurveyDetailViewModel from this HomeViewModel? Does it make sense to make them constraint together at least from the import 💭

File: ios/Surveys/Sources/Presentation/Modules/SurveyDetail/SurveyDetailView.swift [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            }
+            .ignoresSafeArea()
+
+            Rectangle()
+                .foregroundColor(.clear)
+                .background(
+                    LinearGradient(colors: [.clear, .black], startPoint: .top, endPoint: .bottom)
+                )
+                .opacity(0.6)
+                .ignoresSafeArea()
```
Comment (luongvo) [inline ios/Surveys/Sources/Presentation/Modules/SurveyDetail/SurveyDetailView.swift:38]: Should we extract this logic into a reusable view? I see we're using the same UI logic to show the background with a vertical linear gradient from [HomeSurveysView.swift](https://github.com/suho/kmm-ic/pull/127/files#diff-f92f1eacf0e3a425f4a3d60206c3a9d1e4c5ea4f4ac250341a8b41da2fa58330)

**Review by luongvo — COMMENTED** [review]
rest lgtm

File: ios/Surveys/Sources/Presentation/Modules/Home/HomeViewModel.swift (hunk 2 above)
Comment (suho) [reply ios/Surveys/Sources/Presentation/Modules/Home/HomeViewModel.swift:39]: @luongvo Currently, we are creating the next screen's view model from the previous screen's view model. The idea is more supportive of connecting between screens in this case. We will use something like this
```swift
let nextViewModel = NextViewModel()
nextViewModel.delegate = self
return nextViewModel
```

Moreover, in iOS, we also creating different view models for subviews as well: You can check this: https://github.com/nimblehq/mvvm-rxswift-demo/blob/develop/MVVMRxSwiftDemo/Sources/Presentation/Modules/Home/HomeViewModel.swift#L42

>Does it make sense to make them constraint together at least from the import 💭
I'm not sure I get this point, can you give me more detail?

File: ios/Surveys/Sources/Presentation/Modules/SurveyDetail/SurveyDetailView.swift (hunk 3 above)
Comment (suho) [reply ios/Surveys/Sources/Presentation/Modules/SurveyDetail/SurveyDetailView.swift:38]: @luongvo After @nkhanh44 [comment](https://github.com/suho/kmm-ic/pull/127#pullrequestreview-1235294390), I prefer not to extract the whole background view (include the image), but will extract for the gradient view only :D

Comment (suho) [reply ios/Surveys/Sources/Presentation/Modules/SurveyDetail/SurveyDetailView.swift]: Fixed in https://github.com/suho/kmm-ic/pull/127/commits/a775351590ca92cbc7eb82edde34450560b43d11

File: ios/Surveys/Sources/Presentation/Modules/Home/HomeViewModel.swift (hunk 2 above)
Comment (luongvo) [reply ios/Surveys/Sources/Presentation/Modules/Home/HomeViewModel.swift:39]: @suho Thanks for your explanation. Good to know the different approach in the iOS platform 🤗

**Review by luongvo — APPROVED** [review]

**Pull request merged**